src/osip-simulation-controller/src/main/java/edu/kit/pse/osip/simulation/controller/SimulationController.java

```diff
@@ -1,68 +1,244 @@
+            cont.underflowAlarm = new FillAlarm(cont.tank, 5f, AlarmBehavior.SMALLER_THAN);
+            cont.overheatAlarm = new TemperatureAlarm(cont.tank, 5000f, AlarmBehavior.GREATER_THAN);
+            cont.undercoolAlarm = new TemperatureAlarm(cont.tank, 200f, AlarmBehavior.SMALLER_THAN);
+        }
+
+        mixCont.overflowAlarm = new FillAlarm(mixCont.tank, 95f, AlarmBehavior.GREATER_THAN);
+        mixCont.underflowAlarm = new FillAlarm(mixCont.tank, 5f, AlarmBehavior.SMALLER_THAN);
+        mixCont.overheatAlarm = new TemperatureAlarm(mixCont.tank, 5000f, AlarmBehavior.GREATER_THAN);
+        mixCont.undercoolAlarm = new TemperatureAlarm(mixCont.tank, 200f, AlarmBehavior.SMALLER_THAN);
     }
+
+    private void setupServer() throws UaException {
+        int defaultPort = DEFAULT_PORT;
+        for (TankContainer cont : tanks) {
+            cont.server = new TankServer(settingsWrapper.getServerPort(cont.selector, defaultPort++));
+        }
+        mixCont.server = new MixTankServer(defaultPort);
+    }
+
+    private void reSetupServer() {
+        int defaultPort = DEFAULT_PORT;
+        for (TankContainer cont : tanks) {
+            TankServer old = cont.server;
+            try {
+                cont.server = new TankServer(settingsWrapper.getServerPort(cont.selector, defaultPort++));
+            } catch (UaException ex) {
+                System.err.println("Couldn't change OPC UA server port: " + ex.getMessage());
+                cont.server = old;
+            }
+        }
+        MixTankServer old = mixCont.server;
+        try {
+            mixCont.server = new MixTankServer(defaultPort);
+        } catch (UaException ex) {
+            System.err.println("Couldn't change OPC UA server port: " + ex.getMessage());
+            mixCont.server = old;
+        }
+    }
+
+    private void setupView() {
+        Stage help = new HelpDialog();
+        Stage about = new AboutDialog();
+        settingsInterface = new SimulationSettingsWindow(settingsWrapper);
+        controlInterface = new SimulationControlWindow(productionSite);
+
+        currentSimulationView.setOverflowClosedHandler(actionEvent -> productionSite.reset());
+        currentSimulationView.setSettingsButtonHandler(actionEvent -> settingsInterface.show());
+        currentSimulationView.setControlButtonHandler(actionEvent -> controlInterface.show());
+        currentSimulationView.setAboutButtonHandler(actionEvent -> about.show());
+        currentSimulationView.setHelpButtonHandler(actionEvent -> help.show());
+
+        controlInterface.setValveListener((pipe, number) -> pipe.setValveThreshold(number.byteValue()));
+        controlInterface.setTemperatureListener((tankSelector, number) ->
+                productionSite.setInputTemperature(tankSelector, number.byteValue()));
+        controlInterface.setMotorListener(rpm -> productionSite.getMixTank().getMotor().setRPM(rpm));
+        settingsInterface.setSettingsChangedListener(this::reSetupServer);
```

Comment:
Die Settings speichern es momentan noch nicht automatisch in den Wrapper ab, soweit ich weiß.